WordPress/gutenberg
# Unify DataViews Header Title & Subtitle

## What?

This PR achieves the following two things in the header of the Patterns page:

### Truncate the title when it is extremely long

#### Before

![image](https://github.com/WordPress/gutenberg/assets/54422211/7498a18d-b53f-44df-b140-d9a2630f18d0)

#### After

![image](https://github.com/WordPress/gutenberg/assets/54422211/fb5147ec-3cbb-45be-976f-474fa9189499)


### Add a 4px gap between the title and subtitle, which makes the add button and subtitle less cramped

#### Before

![image](https://github.com/WordPress/gutenberg/assets/54422211/cff24ac9-4711-4e07-9478-dbd3364920b8)

#### After

![image](https://github.com/WordPress/gutenberg/assets/54422211/7ca92eda-fedf-4dde-9898-87d0146ec695)

## Testing Instructions

- Visit the Patterns page
- Check all template parts, all patterns, template part category, and pattern category layouts.
